### PR TITLE
[iam] Bind OIDC Client config service

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Bind an unimplemented OIDC Client Config Service to the IAM server

Includes a couple of drive-by fixes:
* Registration of handlers
* Moves loading of test config into the oidc package

## Related Issue(s)

Fixes #

## How to test

CI passes

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: components/iam/pkg/server/server.go [hunk 1]
```diff
@@ -6,30 +6,22 @@ package server
...
 	logger.WithField("config", cfg).Info("Starting IAM server.")
 
-	_, err := db.Connect(db.ConnectionParams{
-		User:     os.Getenv("DB_USERNAME"),
-		Password: os.Getenv("DB_PASSWORD"),
-		Host:     net.JoinHostPort(os.Getenv("DB_HOST"), os.Getenv("DB_PORT")),
-		Database: "gitpod",
-		CaCert:   os.Getenv("DB_CA_CERT"),
-	})
+	_, err := db.Connect(db.ConnectionParamsFromEnv())
```
Comment (easyCZ) [inline components/iam/pkg/server/server.go:24]: Drive-by cleanup.

File: components/iam/pkg/server/server.go [hunk 2]
```diff
@@ -66,33 +56,23 @@ func Start(logger *logrus.Entry, version string, cfg *config.ServiceConfig) erro
...
 }
 
-func registerRootRouter(srv *baseserver.Server) (*chi.Mux, error) {
+type dependencies struct {
+	oidcClientConfigSvc v1.OIDCServiceServer
+
+	oidcService *oidc.Service
+}
+
+func register(srv *baseserver.Server, deps *dependencies) error {
```
Comment (easyCZ) [inline components/iam/pkg/server/server.go:65]: Regroupes registration under a single function, with the handler deps.

File: components/iam/pkg/server/server.go [hunk 3]
```diff
@@ -66,33 +56,23 @@ func Start(logger *logrus.Entry, version string, cfg *config.ServiceConfig) erro
...
 
+	rootHandler.Mount("/oidc", oidc.Router(deps.oidcService))
+
+	// All root requests are handled by our router
 	srv.HTTPMux().Handle("/", rootHandler)
-	return rootHandler, nil
-}
 
-// TODO(at) remove the demo config after start sync'ing with DB
-func loadTestConfig(s *oidc.Service, cfg *config.ServiceConfig) error {
```
Comment (easyCZ) [inline components/iam/pkg/server/server.go:77]: Moved to the `oidc` package, to be co-located with the service.

File: components/iam/pkg/server/server.go [hunk 4]
```diff
@@ -43,20 +35,18 @@ func Start(logger *logrus.Entry, version string, cfg *config.ServiceConfig) erro
 		return fmt.Errorf("failed to initialize IAM server: %w", err)
 	}
 
-	// All root requests are handled by our router
-	rootHandler, err := registerRootRouter(srv)
+	oidcService, err := oidc.NewServiceWithTestConfig(cfg.OIDCClientsConfigFile)
```
Comment (AlexTugarev) [inline components/iam/pkg/server/server.go:38]: I'd prefer to have 2 steps:
1. create service
2. load test config

With DB connection in place, we actually do no longer need the test config.

**Review by AlexTugarev — APPROVED** [review]

File: components/iam/pkg/server/server.go (hunk 4 above)
Comment (easyCZ) [reply components/iam/pkg/server/server.go:38]: What's the difference in those two steps? The function literally does that - create the service, then add the test config

Comment (AlexTugarev) [reply components/iam/pkg/server/server.go:38]: The difference is, that now the service creation in server depends on some config file for tests. 

Anyways, it's temporary. 
So look ahead, I good small step forward would be to introduce a config provider as a dependency to the oidc service. Why? We could start out with backing it by the demo config, but the interesting part is to get to the configs from DB. And in tests we'd somehow mock this provider to not use DB.

**Comment by AlexTugarev** [comment]
Something is wrong with go mod.
Any idea, I saw the same in parallel.

File: components/iam/pkg/server/server.go (hunk 4 above)
Comment (easyCZ) [reply components/iam/pkg/server/server.go:38]: > depends on some config file for tests.

I'm not sure I follow. Do you have an example?

Comment (easyCZ) [reply components/iam/pkg/server/server.go:38]: > And in tests we'd somehow mock this provider to not use DB.

We'd just load the config into the DB, like we do in other tests. We already run MySQL as part of CI (and locally) so this should work normally.

**Review by ArthurSens — APPROVED** [review]
